crates/bevy_ecs/src/system/commands/command_queue.rs

```diff
@@ -9,8 +9,11 @@ struct CommandMeta {
     /// SAFETY: The `value` must point to a value of type `T: Command`,
     /// where `T` is some specific type that was used to produce this metadata.
     ///
+    /// `world` is optional to allow this one function pointer to perform double-duty as a drop.
+    ///
     /// Returns the size of `T` in bytes.
-    apply_command_and_get_size: unsafe fn(value: OwningPtr<Unaligned>, world: &mut World) -> usize,
+    consume_command_and_get_size:
+        unsafe fn(value: OwningPtr<Unaligned>, world: &mut Option<&mut World>) -> usize,
```

Comment:
Using `&mut Option<&mut World>` to allow for the return of the `Option<&mut World>` back to the `CommandQueue` after `consume_command_and_get_size` is done with it.

Reply:
You can avoid the extra `&mut` by reborrowing the `&mut World` before calling `consume_command_and_get_size`, for example by passing `world.as_deref_mut()`

Reply:
I've reduced the usage of the `&mut Option<&mut World>`, but I can't fully eliminate it. `CommandQueue` itself is only passed a mutable reference to `World`, and I don't see a way to convert `&mut World` into `&mut Option<World>`. In fact, I think such a transform is nonsensical.